File: lotto.js
// lotto function
function lotto(howMany, outOf) {
    // assign all numbers from 1..outOf
    const numbers = [];
    for (i = 1; i < (outOf + 1); i++) {
    numbers.push(i);
    };

    // pick lucky numbers
    const luckyNumbers = [];
    for (i = 1; i < (howMany + 1); i++) {
    randomNumber = Math.floor(Math.random() *((outOf + 1)-i));
    luckyNumbers.push(numbers[randomNumber]);
    numbers.splice(numbers.indexOf(numbers[randomNumber]), 1);
    }

    // sort lucky numbers
    const luckyNumbersSorted = luckyNumbers.sort(function(a, b){return a - b});
    return luckyNumbersSorted; // output lucky numbers sorted
};

for (let j = 1; j < 4; j++) {
    console.log(lotto(6, 45)); // call lotto function with 6 aus 45
};

File: lotto.swift
import Foundation

func lotto(_ howMany: Int, outOf: Int) -> [Int] {
    // Create an array of numbers from 1 to outOf
    var numbers = Array(1...outOf)
    
    // Array to hold the lucky numbers
    var luckyNumbers = [Int]()
    
    for _ in 1...howMany {
        // Ensure we don't go out of bounds
        if numbers.isEmpty {
            break
        }
        let randomIndex = Int(arc4random_uniform(UInt32(numbers.count)))
        let luckyNumber = numbers.remove(at: randomIndex)
        luckyNumbers.append(luckyNumber)
    }
    
    // Sort the lucky numbers
    return luckyNumbers.sorted()
}

// Call the lotto function three times
for _ in 1...3 {
    let result = lotto(6, outOf: 45)
    print(result)
}
